test/org/traccar/ProtocolTest.java

```diff
@@ -192,6 +192,16 @@ private void verifyDecodedPosition(Object decodedObject, boolean checkLocation,
             Assert.assertTrue(attributes.get(Position.KEY_FUEL_LEVEL) instanceof Number);
         }
 
+        if (attributes.containsKey(Position.KEY_BATTERY)) {
+            Assert.assertTrue(attributes.get(Position.KEY_BATTERY) instanceof Number);
+        }
+
+        if (attributes.containsKey(Position.KEY_BATTERY_LEVEL)) {
+            Object batteryLevel = attributes.get(Position.KEY_BATTERY_LEVEL);
```

Comment:
I think you can just downcast to Number and get int value here. Instanceof check is not necessary because it will fail with pretty obvious error anyway.